ledger/participant-integration-api/src/main/scala/platform/index/InMemoryStateUpdater.scala

```diff
@@ -44,12 +46,21 @@ final class InMemoryStateUpdater(
         Update.CommandRejected,
     ) => TransactionLogUpdate.TransactionRejected,
     updateLedgerEnd: (Offset, Long) => Unit,
+    updatePackageMetadata: Update.PublicPackageUpload => Unit,
 ) {
 
+  val packageMetadataFlow: Sink[(Vector[(Offset, Update)], Long), Future[Done]] = Sink.foreach {
+    case (batch, _) =>
+      batch
+        .collect { case (_, event: Update.PublicPackageUpload) => event }
+        .foreach(updatePackageMetadata)
```

Comment:
We are running here expensive computation on the akka default scheduler. That is not good, we want to run the computation on an appropriate and dedicated execution context